File: Blind75/02_Two_Pointers/02_Two_Sum_II_-_Input_Array_Is_Sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        i = 0
        j = len(numbers) - 1

        while i < j:
            val = numbers[i] + numbers[j]
            if val == target:
                return [i+1,j+1]
            if val < target:
                i += 1
            elif val > target:
                j -= 1
        
        return []

File: 02_Two_Pointers/02_Two_Sum_II_-_Input_Array_Is_Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0, j = numbers.length - 1;

        while (i < j) {
            int value = numbers[i] + numbers[j];

            if (value == target)
                return new int[] { i + 1, j + 1 };
            if (value < target)
                i++;
            if (value > target)
                j--;
        }

        return new int[0];
    }
}